operators/dream_texture.py

```diff
@@ -38,10 +38,24 @@ def invoke(self, context, event):
         return {'CANCELLED'}
 
     def execute(self, context):
-        history_entry = context.preferences.addons[StableDiffusionPreferences.bl_idname].preferences.history.add()
-        for prop in context.scene.dream_textures_prompt.__annotations__.keys():
-            if hasattr(history_entry, prop):
-                setattr(history_entry, prop, getattr(context.scene.dream_textures_prompt, prop))
+        history_entries = []
+        is_file_batch = context.scene.dream_textures_prompt.prompt_structure == file_batch_structure.id
+        file_batch_lines = []
+        if is_file_batch:
+            file_batch_lines = [line for line in context.scene.dream_textures_prompt_file.lines if len(line.body.strip()) > 0]
+            history_entries = [context.preferences.addons[StableDiffusionPreferences.bl_idname].preferences.history.add() for _ in file_batch_lines]
+        else:
+            history_entries = [context.preferences.addons[StableDiffusionPreferences.bl_idname].preferences.history.add() for _ in range(context.scene.dream_textures_prompt.iterations)]
+        for i, history_entry in enumerate(history_entries):
+            for prop in context.scene.dream_textures_prompt.__annotations__.keys():
+                if hasattr(history_entry, prop):
+                    setattr(history_entry, prop, getattr(context.scene.dream_textures_prompt, prop))
+            if is_file_batch:
+                history_entry.prompt_structure = custom_structure.id
+                history_entry.prompt_structure_token_subject = file_batch_lines[i].body
+        
+        if is_file_batch:
+            context.scene.dream_textures_prompt.iterations = 1
```

Comment:
Iterations should be set to 1 before batch history entries are made, otherwise recalling one will include iterations that were not previously used.
```suggestion
        if is_file_batch:
            context.scene.dream_textures_prompt.iterations = 1
            file_batch_lines = [line for line in context.scene.dream_textures_prompt_file.lines if len(line.body.strip()) > 0]
            history_entries = [context.preferences.addons[StableDiffusionPreferences.bl_idname].preferences.history.add() for _ in file_batch_lines]
        else:
            history_entries = [context.preferences.addons[StableDiffusionPreferences.bl_idname].preferences.history.add() for _ in range(context.scene.dream_textures_prompt.iterations)]
        for i, history_entry in enumerate(history_entries):
            for prop in context.scene.dream_textures_prompt.__annotations__.keys():
                if hasattr(history_entry, prop):
                    setattr(history_entry, prop, getattr(context.scene.dream_textures_prompt, prop))
            if is_file_batch:
                history_entry.prompt_structure = custom_structure.id
                history_entry.prompt_structure_token_subject = file_batch_lines[i].body
```